src/Refactoring-Core/RBAddClassRefactoring.class.st

```diff
@@ -39,31 +51,61 @@ RBAddClassRefactoring class >> model: aRBSmalltalk addClass: aName superclass: a
 		yourself
 ]
 
+{ #category : #'instance creation' }
+RBAddClassRefactoring class >> model: aRBSmalltalk addClass: aName superclass: aClass subclasses: aCollection traits: traitsCollection category: aSymbol [
+
+	^ self new
+		  model: aRBSmalltalk;
+		  addClass: aName
+		  superclass: aClass
+		  subclasses: aCollection
+		  traits: traitsCollection
+		  category: aSymbol;
+		  yourself
+]
+
 { #category : #initialization }
-RBAddClassRefactoring >> addClass: aName superclass: aClass subclasses: aCollection category: aSymbol [ 
+RBAddClassRefactoring >> addClass: aName superclass: aClass subclasses: aCollection category: aSymbol [
+
 	self className: aName.
 	superclass := self classObjectFor: aClass.
-	subclasses := aCollection collect: [:each | self classObjectFor: each].
+	subclasses := aCollection collect: [ :each | 
+		              self classObjectFor: each ].
+
+	traits := #(  ).
+	category := aSymbol
+]
+
+{ #category : #initialization }
+RBAddClassRefactoring >> addClass: aName superclass: aClass subclasses: subclassesCollection traits: traitsCollection category: aSymbol [
+
+	self className: aName.
```

Comment:
there is code duplication here.
You should call `self addClass: aName superclass: aClass subclasses: aCollection category: aSymbol` and then only initialize traits i.v.